pkg/build/controller/controller.go

```diff
@@ -326,7 +323,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 
 	// Update the build object when it progress to a next state or the reason for
 	// the current state changed.
-	if build.Status.Phase != nextStatus && !buildutil.IsBuildComplete(build) {
+	if (!hasBuildPodNameAnnotation(build) || build.Status.Phase != nextStatus) && !buildutil.IsBuildComplete(build) {
```

Comment:
oh, well i see how that code works now, it sets the new status and then checks if it's complete, so that makes sense.